### ISSUE TITLE
Snap apps icons (in the task bar) are missing in some snaps.

### ISSUE THREAD

**Issue by chifreagle** [budgie-desktop/budgie-desktop#1180]
**Operating System**: Solus updated 3 october 
**Budgie version**: Budgie 1.4 updated 

**The issue encountered**
Snap apps icons (in the task bar) are missing in some snaps.
**Steps to reproduce the issue**
1. download a snap (gravit designer as example)
2. start snap
-> the task bar uses a weird blue window icon instead of the snap's icon

**Comment by ikeydoherty** [comment]
Are the pin/unpin actions present?

**Comment by chifreagle** [comment]
no

**Comment by ikeydoherty** [comment]
OK so its more likely the `.desktop` file is missing. Are the actions present post-reboot?

**Comment by chifreagle** [comment]
rebooting adds an entry in budgie menu issue still present

**Comment by chifreagle** [comment]
![screenshot from 2017-10-03 13 20 44](https://user-images.githubusercontent.com/11044024/31124765-c2d103a8-a83d-11e7-89c8-c7f21fee5086.png)

the menu entry shows correct icon and name in the wrong category 
taskbar icon not present

**Comment by TheMarketAnarchist** [comment]
I had an issue with this with the Brave snap. The issue is, as Ikey mentioned, an incorrect ```.desktop``` file. Here is a basic solution:

Make sure the .desktop name matches your app's WM_Class(string).

If you don't know how to get that, follow these steps:

1. Install xprop: sudo eopkg it xprop
2. Run in terminal: xprop
3. Click the window you want to get their "WM_Class(string)" and look for the name. It's usually at the bottom side
4. Rename your .desktop, open the app, and right click to pin it.
5. Profit

Taken from the top [comment](https://www.reddit.com/r/SolusProject/comments/6ykl6u/ability_to_pin_snaps_to_panel_budgie/) on my reddit post asking the same question!

**Comment by n8v8R** [comment]
Distro: Solus 3.1 (3.26.1)
Kernel: 4.13.8-28.current x86_64
bits: 64
gcc: 6.4.0
Desktop: Budgie 10.4
Budgie version: Budgie 1.4 updated

The issue encountered
'pin to panel' is missing for snap app Brave browser

> _NET_WM_DESKTOP(CARDINAL) = 0
> _NET_WM_ICON(CARDINAL) = 
> WM_WINDOW_ROLE(STRING) = "browser-window"
> WM_CLASS(STRING) = "brave", "brave"

**Comment by VinnyFonseca** [comment]
I may have found something else. This is not snap based, but it is related to `~/.local/share/applications` `.desktop` files.

Nemo automatically renames the `.desktop` files superficially (apparently, only visually) based on the `Name[en_GB]` property (or any other localisation), but the real filename doesn't really change.

Pin to panel wasn't working until I went to Nautilus, saw that the filename was still `Avocode (copy).desktop` as I had copied another `.desktop` file and amended it as I wouldn't remember the whole syntax.

Once I renamed it to `Trello.desktop` as indicated on xprop, pin to panel started working again. So just a reminder to check the `.desktop` `Name` properties, and when renaming these files do it through something guaranteed, such as terminal.

**Comment by pappfer** [comment]
I had the same problem with Slack snap. I had to add `StartupWMClass=Slack` to `/var/lib/snapd/desktop/applications/slack_slack.desktop` and that fixed it.

**Comment by MariusVB** [comment]
Had the same issue, but fixed it by following this:

> I had the same problem with Slack snap. I had to add `StartupWMClass=Slack` to `/var/lib/snapd/desktop/applications/slack_slack.desktop` and that fixed it.

But the icon in the workspace viewer in task-bar and in the icon in the notification area are still blank.
Image here: https://i.imgur.com/C2NhN4T.png - The slack icon is the black box next to the smiley

Any idea what to do here?

**Comment by pappfer** [comment]
> Any idea what to do here?
Yes, this is still an issue. What I did was, I deleted the Slack snap and installed it via Flatpak, now I have a status icon.

**Comment by MariusVB** [comment]
> > Any idea what to do here?
> 
> Yes, this is still an issue. What I did was, I deleted the Slack snap and installed it via Flatpak, now I have a status icon.

Thanks - fixed the issue!

**Comment by FeryET** [comment]
I have the same issue with Pycharm and Gimp in KDE. When I search for Pycharm in the search bar this shows up:

 
![image](https://user-images.githubusercontent.com/30388648/91525135-0fb08380-e916-11ea-8124-3f7698f0fa23.png)


But after running the application my taskbar looks like this. I've hovered over what supposed to be Pycharm's icon, and the preview text is displayed, but there's no icon:

![image](https://user-images.githubusercontent.com/30388648/91525175-27880780-e916-11ea-9a70-75665f3112ef.png)
